### PR TITLE
This commit contains some small changes to catch blocks and tests

### PR DESCRIPTION
**Pull request by dshirt**
Made very small changes to catch block in Employee constructor. The catch had thrown a new exception before and now prints a statement. new test wrote for this.
another catch also changed in employee class and test written for same reason.
new method writtten for invalidDOBformatt in Employeefactory.

### CODE REVIEW

File: src/test/java/employee/com/CommissionBasedStaffTest.java [hunk 1]
```diff
@@ -15,93 +16,96 @@
...
     @DisplayName("********Testing the CommissionBasedEmployee EmployeeID Not A Number**********")
     @Test
-    void testCommissionBasedStaffEmployeeIdNotANumber() {
-        final String mustBeNum = "EmployeeID is not a valid number";
-        Exception notANum = assertThrows(NumberFormatException.class,()->
-                EmployeeFactory.getValidCommissionBasedStaff().setEmployeeId("AAA"));
-        assertEquals(mustBeNum, notANum.getMessage());
+    void testCommissionBasedStaffEmployeeIdNotANumber()  {
+        EmployeeFactory.getValidCommissionBasedStaff().setEmployeeId("AAA");
     }
```
Comment (daniel-fitzg) [inline src/test/java/employee/com/CommissionBasedStaffTest.java]: Is this change no longer testing anything? Not seeing any assert statements.

File: src/test/java/employee/com/CommissionBasedStaffTest.java [hunk 2]
```diff
@@ -15,93 +16,96 @@
...
-    void testValidDOBOfEmployee() throws ParseException {
-        assertEquals("30/07/1996", EmployeeFactory.getValidCommissionBasedStaff().getDOB());
+    void testValidDOBOfEmployee()  {
+        assertEquals("30/07/1996", getValidCommissionBasedStaff().getDOB());
+    }
+
+    @DisplayName("********Testing invalid format for dob in constructor")
+    @Test()
+    void testInvalidDOBformat() {
+        EmployeeFactory.getInvalidEmployeDOBFormat();
```
Comment (daniel-fitzg) [inline src/test/java/employee/com/CommissionBasedStaffTest.java]: Are you testing anything in here? No asserts.

File: src/main/java/employee/com/Employee.java [hunk 3]
```diff
@@ -27,7 +27,7 @@ public Employee(String employeeId, String name, String address, String dob, Stri
         try {
             setDOB(dob);
         } catch (ParseException e) {
-            e.printStackTrace();
+          System.out.println("Employee date of birth must be in format dd/MM/yyyy");
```
Comment (rdmoraes) [inline src/main/java/employee/com/Employee.java]: Why did you change it to output stream?

File: src/test/java/employee/com/EmployeeFactory.java [hunk 4]
```diff
@@ -21,6 +21,11 @@ static CommissionBasedStaff getValidCommissionBasedStaff() {
                 "30/07/1996", "1234567G", 20000, 0.1);
     }
 
+    static CommissionBasedStaff getInvalidEmployeDOBFormat() {
```
Comment (daniel-fitzg) [inline src/test/java/employee/com/EmployeeFactory.java:24]: Could you change the name of the method to:
getComissionBasedStaffWithInvalidDOB()

I think it makes more sense.

**Review by daniel-fitzg — CHANGES_REQUESTED** [review]

File: src/test/java/employee/com/CommissionBasedStaffTest.java [hunk 5]
```diff
@@ -15,93 +16,96 @@
...
 
     @DisplayName("********Testing the CommissionBasedEmployee EmployeeID Not A Number**********")
     @Test
-    void testCommissionBasedStaffEmployeeIdNotANumber() {
-        final String mustBeNum = "EmployeeID is not a valid number";
-        Exception notANum = assertThrows(NumberFormatException.class,()->
-                EmployeeFactory.getValidCommissionBasedStaff().setEmployeeId("AAA"));
-        assertEquals(mustBeNum, notANum.getMessage());
+    void testCommissionBasedStaffEmployeeIdNotANumber()  {
+        EmployeeFactory.getValidCommissionBasedStaff().setEmployeeId("AAA");
```
Comment (rdmoraes) [inline src/test/java/employee/com/CommissionBasedStaffTest.java]: As you import  **employee.com.EmployeeFactory.getValidCommissionBasedStaff** (line 3) I would change to **employee.com.EmployeeFactory.*** and rewrite this line as _getValidCommissionBasedStaff().setEmployeeID("AAA")_

File: src/test/java/employee/com/CommissionBasedStaffTest.java (hunk 2 above)
Comment (rdmoraes) [inline src/test/java/employee/com/CommissionBasedStaffTest.java]: Same here **EmployeeFactory.getInvalidEmployeeDOBFormat()** to **getInvalidEmployeeDOBFormat()**

File: src/test/java/employee/com/EmployeeFactory.java (hunk 4 above)
Comment (rdmoraes) [inline src/test/java/employee/com/EmployeeFactory.java]: Minor typo missing letter 'e' in employee in getInvalidEmploye***e***DOBFormat()

**Review by rdmoraes — CHANGES_REQUESTED** [review]
Minor changes required

File: src/main/java/employee/com/Employee.java [hunk 6]
```diff
@@ -15,12 +19,11 @@
     private String address;
     private String dob;
     private String ppsNo;
-    private final int EMPLOYEE_ID_MIN_LEN= 2;
-    private final int PPS_NOT_MIN_LEN = 8;
-    private final int MIN_AGE = 16;
-    private final int MIN_NAME_LENGTH = 3;
+    EmployeeValidationService employeeValidationService =
+            new EmployeeValidationService();
```
Comment (dshirt) [inline src/main/java/employee/com/Employee.java]: This is the only way i could make this work if it was moved into constructor it broke the code

File: src/main/java/employee/com/DepartmentManager.java [hunk 7]
```diff
@@ -1,5 +1,7 @@
 package employee.com;
 
+import java.text.ParseException;
```
Comment (daniel-fitzg) [inline src/main/java/employee/com/DepartmentManager.java]: Unused import statement.

File: src/main/java/employee/com/Director.java [hunk 8]
```diff
@@ -1,12 +1,14 @@
 package employee.com;
 
+import java.text.ParseException;
```
Comment (daniel-fitzg) [inline src/main/java/employee/com/Director.java]: Unused import.

File: src/main/java/employee/com/Employee.java [hunk 9]
```diff
@@ -1,11 +1,15 @@
...
+* RENAN Leite De Moraes
+*  Email: g00351835@gmit.ie
+*  Date: 28/April/2020
+*
+ */
 
+package employee.com;
 
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
```
Comment (daniel-fitzg) [inline src/main/java/employee/com/Employee.java]: Unused imports.

File: src/main/java/employee/com/Employee.java [hunk 10]
```diff
@@ -15,20 +19,15 @@
     private String address;
     private String dob;
     private String ppsNo;
-    private final int EMPLOYEE_ID_MIN_LEN= 2;
-    private final int PPS_NOT_MIN_LEN = 8;
-    private final int MIN_AGE = 16;
-    private final int MIN_NAME_LENGTH = 3;
+    EmployeeValidationService employeeValidationService =
```
Comment (daniel-fitzg) [inline src/main/java/employee/com/Employee.java]: Can be private.

File: src/main/java/employee/com/Manager.java [hunk 11]
```diff
@@ -1,11 +1,13 @@
 package employee.com;
 
+import java.text.ParseException;
```
Comment (daniel-fitzg) [inline src/main/java/employee/com/Manager.java]: Unused import.

**Review by daniel-fitzg — CHANGES_REQUESTED** [review]
Just a few little things left.

File: src/main/java/employee/com/Employee.java (hunk 6 above)
Comment (daniel-fitzg) [reply src/main/java/employee/com/Employee.java]: @dshirt I just meant the variable could be private instead of package-private

**Review by daniel-fitzg — APPROVED** [review]
LGTM

**Review by rdmoraes — APPROVED** [review]
Looks good to me as well! Are we back with the validation class again?

**Comment by daniel-fitzg** [comment]
@rdmoraes I think the setters were just getting too big and messy. Moving into it's own  class seemed a cleaner solution.

**Pull request merged**